src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/IPropertyExtensions.cs

```diff
@@ -28,12 +28,12 @@ internal static IPropertyMetadata ToPropertyMetadata(this IProperty property, Ty
             propertyMetadata.IsForeignKey = property.IsForeignKey();
 
             propertyMetadata.IsEnum = property.ClrType.GetTypeInfo().IsEnum;
-            propertyMetadata.IsReadOnly = property.IsReadOnlyBeforeSave;
+            propertyMetadata.IsReadOnly = property.BeforeSaveBehavior != PropertyValueBehavior.UseValue;
```

Comment:
`Readonly` is used to determine whether to create an editable field for the property or not